### ISSUE TITLE
Change report location of prefer-const to that of the `let` keyword

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#13167]
**What rule do you want to change?**
prefer-const

**Does this change cause the rule to produce more or fewer warnings?**
No change

**How will the change be implemented? (New option, new default behavior, etc.)?**
New report location

**Please provide some example code that this change will affect:**

```js
/*eslint prefer-const: 2*/

let foo = 3;
```
https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgcHJlZmVyLWNvbnN0OiAyKi9cblxubGV0IGZvbyA9IDNcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OCwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==

**What does the rule currently do for this code?**
Highlights `foo`.
**What will the rule do after it's changed?**
Highlights `let`.

Reasons for this change:
1. The token to be highlighted should match the token to be fixed.
2. Current highlight has too many possible interpretations. It can mean `foo` is unused (no-unused-var) , or `foo` is redeclared (parse error).

**Are you willing to submit a pull request to implement this change?**
Sure

**Comment by anikethsaha** [comment]
IMO, the message says - `'foo' is never reassgined` -  which I think does make sense to locate at `foo`.

**Comment by mdjermanovic** [comment]
I agree with the previous comment. Since this rule checks and reports each variable individually, the identifier seems to be the correct location to report.

**Comment by golopot** [comment]
Closing because I agree with https://github.com/eslint/eslint/issues/13167#issuecomment-611378996.
